superset-frontend/src/dashboard/util/permissionUtils.test.ts

```diff
@@ -132,22 +139,38 @@ test('isUserAdmin returns false for non-admin user', () => {
   expect(isUserAdmin(ownerUser)).toEqual(false);
 });
 
-test('canUserAccessSqlLab returns true for admin user', () => {
-  expect(canUserAccessSqlLab(adminUser)).toEqual(true);
-});
-
-test('canUserAccessSqlLab returns false for undefined', () => {
-  expect(canUserAccessSqlLab(undefined)).toEqual(false);
+test('userHasPermission always returns true for admin user', () => {
+  arbitraryPermissions.forEach(permissionView => {
+    expect(
+      userHasPermission(adminUser, permissionView[1], permissionView[0]),
+    ).toEqual(true);
+  });
 });
 
-test('canUserAccessSqlLab returns false for undefined user', () => {
-  expect(canUserAccessSqlLab(undefinedUser)).toEqual(false);
+test('userHasPermission always returns false for undefined user', () => {
+  arbitraryPermissions.forEach(permissionView => {
+    expect(
+      userHasPermission(undefinedUser, permissionView[1], permissionView[0]),
```

Comment:
A minor nit, and I agree that the order of first having the view name, and then the perm makes sense, but from a usability perspective, might it be less error prone to just ask for the `permissionView` as the second arg to the `userHasPermission` function? Like so:
```javascript
const userHasPermission = userHasPermission(underfinedUser, permissionView);
```

Especially, as both `viewName` and `permissionName` are both `string`, making it easy to mix them up. Having a tuple `[string, string]` feels less error prone.

Reply:
I also toyed with a much less error prone interface like `{view: "Something", permission: "can_read"}` but that'd be inconsistent with the tuples found on the user, so I don't think that's great either.

Reply:
@giftig you're right, there's probably no super safe way to do this. However, I assume perms usually come back in tuple format, so for that reason `[string, string]` could be good. But we can do that later, this is already a really good improvement.